src/lib/file.js

```diff
@@ -161,17 +190,47 @@ export function cacheFile(cache, file) {
         addOriginalRepresentation(file);
     }
 
-    cache.set(file.id, file);
+    // Cache using file ID as a key
+    cache.set(`f_${file.id}`, file);
+
+    // Cache using file version ID as key
+    if (file.file_version && file.file_version.id) {
+        cache.set(`fv_${file.file_version.id}`, file);
+    }
 }
 
 /**
- * Wrapper for uncaching a file object.
+ * Wrapper for uncaching a file object. We uncache both the key by file ID and the key by file version ID.
  *
  * @public
  * @param {Cache} cache - Cache instance
  * @param {Object} file - Box file or simple { id: fileId } object
  * @return {void}
  */
 export function uncacheFile(cache, file) {
-    cache.unset(file.id);
+    cache.unset(`f_${file.id}`);
```

Comment:
Instead use

file_123
file_version_123

since thats what is used by our token service, f_ is a bit ambiguous (folder vs file).

also use a getKey() function that does this appending, otherwise hard to refactor and string duplication.